### PR TITLE
Py37

### PR DESCRIPTION
**Pull request by madhuBayy**
@ldko @somexpert 
I have used futurize tool to convert the python2 code of aubreylib to python 3. Please review the code and give me the feedback. Tox tests will fail as this has dependecies on pyuntl which uses urllib2 module.

### CODE REVIEW

File: aubreylib/resource.py
Comment (ldko) [inline aubreylib/resource.py]: What does converting this to a `list` do?

File: tox.ini
Comment (somexpert) [inline tox.ini]: We should take this time to specify the version of Python used with flake8. Can just be this: 
```suggestion
envlist = py36, py37, py37-flake8
```

Comment (somexpert) [reply tox.ini]: You would also need to change the testenv name on line 12, and also in the Travis config to define the version of Python and use the new testenv name.

File: aubreylib/resource.py
Comment (somexpert) [inline aubreylib/resource.py]: Doesn't look like we use `urllib.error` or `urllib.parse`. Need to look through the same imports in other modules.

File: aubreylib/resource.py
Comment (somexpert) [inline aubreylib/resource.py]: I could be wrong, but isn't this converting the output from `read()`, which I presume is Bytes, to String, then going _back_ to Bytes? _If_ that's the case, then I would think we could just do the following:
```suggestion
    metadata_stringfile = metadata_filehandle.read()
```

Comment (somexpert) [reply aubreylib/resource.py]: But that could be wrong. Need to be sure before we would make such a change.

File: requirements-test.txt
Comment (somexpert) [inline requirements-test.txt]: I think pyuntl should go into a `requirements.txt` file, along with lxml and pypairtree.

Comment (somexpert) [reply requirements-test.txt]: Also, we wouldn't want to keep the python2to3 branch in here, we'd want it to use master.

File: setup.py
Comment (somexpert) [inline setup.py:16]: Why are we removing this from here?

File: aubreylib/resource.py
Comment (ldko) [inline aubreylib/resource.py]: We should watch for comments that should be updated with the code changes.

File: requirements-test.txt
Comment (madhuBayy) [reply requirements-test.txt]: I'm using python2to3 branch for now as the python3 version code is not yet merged into master branch of pyuntl project yet.

File: setup.py
Comment (madhuBayy) [reply setup.py:16]: I tried using the pyuntl dependency from install_requires but it wasn't getting installed properly. So I tried to move it from here to requirements and it worked!

File: aubreylib/resource.py
Comment (ldko) [reply aubreylib/resource.py]: In Python 3, the way things are in `open_system_file`, that read() could return `bytes` or `str` depending if the file is being read with `urlopen` (now bytes) or from the file system with `open` (by default assumes text and returns a `str`). We may want to modify `aubreylib.system.open_system_file` to return a consistent type.

File: aubreylib/resource.py
Comment (madhuBayy) [reply aubreylib/resource.py]: It's a thing the futurize tool does automatically. I missed removing it here.

File: aubreylib/resource.py
Comment (madhuBayy) [reply aubreylib/resource.py]: I can make the `aubreylib.system.open_system_file` function return bytes type and then apply BytesIO on the filehandler, which will be consumed by `pyuntl.untldoc.untlxml2pydict` function.

Comment (ldko) [reply aubreylib/resource.py]: I think `open_system_file` used to always return `str` in Python 2, I haven't looked at all of this, but did you make sure we want to be dealing with `bytes` rather than `str` now?

Comment (madhuBayy) [reply aubreylib/resource.py]: The untlxml2pydict() makes a call to lxml.etree.iterparse(file_name), which according to my understanding should either read the file directly or get the data read as bytes object. I referred to https://stackoverflow.com/questions/46689317/error-while-parsing-lxml

File: aubreylib/resource.py
Comment (ldko) [inline aubreylib/resource.py]: I am thinking this may be another `list()` conversion we don't need.

**Review by ldko — COMMENTED** [review]
I have one more code comment here. Otherwise, I think the code changes look good. Before we are ready to merge this to master (when we are switching all the aubrey stuff over to Python 3 in production) we will need to change the pyuntl dependency.

File: README.md
Comment (ldko) [inline README.md]: I think the `pip install git+git://github.com/unt-libraries/aubreylib` already installs dependencies.

File: README.md
Comment (ldko) [inline README.md]: What do you think about reorganizing this sentence:

`To run the tests and the `flake8` style checker, you can use tox:`

File: README.md
Comment (madhuBayy) [reply README.md]: well...it installs the dependencies automatically, but those used for master which is in py27 version. Actually I'll get rid of the installations from `setup.py` and instead use `requirements.txt` which has the latest version of pyuntl. And also use `requirements-test.txt` before running tox tests.

File: README.md
Comment (madhuBayy) [reply README.md]: sure! I'll do that

File: README.md
Comment (ldko) [reply README.md]: I thought using requirements.txt is only temporary until we we are ready to merge pyuntl for Python 3 into master? If that is the case, I wouldn't bother putting special instructions into the readme for this.

Comment (madhuBayy) [reply README.md]: yes, it is temporary. Makes sense to leave it out, as I just saw there's an issue opened to address pyuntl dependency.

**Comment by ldko** [comment]
I think this is ready to merge into py37 branch. @somexpert do you have further changes?

File: .travis.yml
Comment (somexpert) [inline .travis.yml:9]: So we can tell Travis to run both the 3.7 envs here and save a couple lines:
```suggestion
      env: TOXENV=py37,py37-flake8
```

Comment (somexpert) [reply .travis.yml:9]: Hmmm, although that saves a couple lines here, the test will not show up as a separate environment in Travis. So up to you whether you think it's worth doing it that way or not.

Comment (madhuBayy) [reply .travis.yml:9]: I think I'll leave it the way it is now.

File: aubreylib/system.py
Comment (somexpert) [inline aubreylib/system.py:3]: What about `urllib.parse`?

**Comment by somexpert** [comment]
So I'm still not sure about how the requirements are being specified. I get that there's some issues because we need non-master versions, but how about instead of resorting to duplicating one of the main requirements in two different places (setup.py and requirements.txt), what if we use the same strategy that worked for pyatompub, specifying the github URL in the `install_requires` portion of `setup.py` and then use `pip install .` whenever this needs to be installed?

File: aubreylib/system.py
Comment (ldko) [reply aubreylib/system.py:3]: :/ seems like that should have failed some test? maybe we need to add more tests...

Comment (somexpert) [reply aubreylib/system.py:3]: it doesn't even fail when I import that module... not sure why

Comment (somexpert) [reply aubreylib/system.py:3]: oh right, gotta use one of the classes/functions :D

Comment (somexpert) [reply aubreylib/system.py:3]: but yeah, it's odd that no tests complained

**Comment by madhuBayy** [comment]
well..We can do something similar to pyatompub for installation of pyuntl, but as pyuntl 's `python2to3` branch will get merged to `py3` branch and anyway I opened an issue to address pyuntl dependency and I thought to handle it there.

File: aubreylib/system.py
Comment (ldko) [reply aubreylib/system.py:3]: To note, for some reason this works, but let's import explicitly for clarity:
```
Python 3.7.3 (default, Jun 27 2019, 16:48:23) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib.request
>>> urllib.parse.urlsplit('http://unt.edu/something/one/file.txt')
SplitResult(scheme='http', netloc='unt.edu', path='/something/one/file.txt', query='', fragment='')
```
```
Python 3.7.3 (default, Jun 27 2019, 16:48:23) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> urllib.parse.urlsplit('http://unt.edu/something/one/file.txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'urllib' is not defined
```

**Comment by somexpert** [comment]
It's fine to deal with it in py3, but I think once we get there we should use the same strategy as pyatompub

File: aubreylib/system.py
Comment (madhuBayy) [reply aubreylib/system.py:3]: Hmm..I was looking at the tests if they were covering the file locations with http and https and they are. It seems weird the way `urlsplit` worked without importing `urllib.parse` and just with `urllib.request` but I'll add it to the imports as you suggested.

**Review by somexpert — APPROVED** [review]
LGTM :thumbsup:

**Pull request merged**
